test/functional/feature_evm_token_split.py

```diff
@@ -0,0 +1,768 @@
+            )
+        else:
+            assert_equal(self.nodes[0].getaccount(destination_address), [])
+
+        # Calculate minted balance
+        minted_balance = (Decimal(1000.00000000) - Decimal(str(amount))) / abs(
+            split_multiplier
+        )
+
+        # Limit to 8 decimal places
+        balance_after_split = self.satoshi_limit(minted_balance)
+
+        # Check minted balance
+        assert_equal(
+            self.nodes[0].gettoken("META")[f"{self.meta_final_id}"]["minted"],
+            balance_after_split + expected_balance,
+        )
+
+    def transfer_domain_multiple_split(self):
+
+        # Rollback
+        self.rollback_to(self.block_height)
+
+        # Fund address
+        self.fund_address(self.address, self.evm_address)
+
+        # Split token
+        self.split_token(self.contract_address_metav1, self.contract_address_metav2)
+
+        # Split token twice
+        self.split_token(
+            self.contract_address_metav2, self.contract_address_metav3, "v2"
+        )
+
+        # Generate new destination address
+        destination_address = self.nodes[0].getnewaddress("", "legacy")
+
+        # Transfer out METAv1 tokens
+        self.nodes[0].transferdomain(
+            [
+                {
+                    "src": {
+                        "address": self.evm_address,
+                        "amount": "20@META/v1",
+                        "domain": 3,
+                    },
+                    "dst": {
+                        "address": destination_address,
+                        "amount": "20@META/v1",
+                        "domain": 2,
+                    },
+                    "singlekeycheck": False,
+                }
+            ]
+        )
+        self.nodes[0].generate(1)
+
+        # Check META balance
+        assert_equal(
+            self.nodes[0].getaccount(destination_address)[0], "80.00000000@META"
+        )
+
+        # Check minted balance
+        assert_equal(
+            self.nodes[0].gettoken("META")[f"{self.meta_final_id}"]["minted"],
+            Decimal(4000.00000000),
+        )
+
+    def intrinsic_token_split(self, amount, split_multiplier):
+
+        # Rollback
+        self.rollback_to(self.block_height)
+
+        # Fund address
+        self.fund_address(self.address, self.evm_address)
+
+        # Split token
+        self.split_token(
+            self.contract_address_metav1,
+            self.contract_address_metav2,
+            "v1",
+            split_multiplier,
+        )
+
+        # Execute and test split transaction
+        self.execute_split_transaction(
+            self.contract_address_metav1,
+            self.contract_address_metav2,
+            amount,
+            split_multiplier,
+        )
+
+        # Check minted balance
+        assert_equal(
+            self.nodes[0].gettoken("META")[f"{self.meta_final_id}"]["minted"],
+            Decimal(1960.00000000) + (amount * split_multiplier),
```

Comment:
also check that economy values about the TD are updated correctly. Since its changed from META/v1 to META, the according amounts in the economy need an update. (META/v1 dvm->evm reduces and META dvm->evm increased)